examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -742,13 +763,43 @@ def _viz_fetcher(self, post_sim_update_dict):
                 )
             )
 
+    def _reset_current_episode(self):
+        # save initial agent states
+        prev_init_agent_states = self._init_agent_states
+        self._init_agent_states = {}
+
+        # reset current episode
+        # env's current_episode setter will set internal attribute
+        # _episode_from_iter_on_reset to False as a result reset
+        # will be called without the episode changing
+        self._sandbox_service.env.current_episode = self._current_episode
+        self._sandbox_service.end_episode(do_reset=True)
+
+        # restore initial agent states
+        sim = self.get_sim()
+        for agent_idx in range(self.get_num_agents()):
+            articulated_agent = get_articulated_agent(sim, agent_idx)
+            prev_pos, prev_rot = prev_init_agent_states[agent_idx]
+            articulated_agent.base_pos = prev_pos
+            articulated_agent.base_rot = float(prev_rot)
+        self._init_agent_states = prev_init_agent_states
+
+    def _set_next_episode(self):
```

Comment:
I don't understand this helper.

Reply:
The idea is instead of having the code written under 
```
if self._sandbox_service.gui_input.get_key_down(GuiInput.KeyNS.<Any key>):
    # code
``` 
wrap the code in a method and give it a meaningful name. 

For example, having  `self._set_next_episode()` is (IMO) more meaningful than `self._sandbox_service.end_episode(do_reset=True)`. 

But I can roll this back.

Reply:
Ok. I was just thrown by the name. So it means "end the episode and start the next episode". I guess I would call it `_end_episode_and_start_next`.